src/client/executor.rs

```diff
@@ -460,13 +484,21 @@ impl Client {
                     handler.handle_command_failed_event(command_failed_event);
                 });
 
-                if let Some(session) = session {
+                err.add_labels(Some(connection), session, Some(retryability))?;
+
+                if let Some(ref mut session) = session {
                     if err.is_network_error() {
                         session.mark_dirty();
                     }
+
+                    if err.contains_label(TRANSIENT_TRANSACTION_ERROR)
```

Comment:
This probably won't matter in practice, but just in case: we probably should put this logic after the entire `match` in case `op.handle_response` below in the `Ok` branch happens to return an error with one of these labels.